### PR TITLE
Ws/fix structs and concurrency

### PR DESCRIPTION
**Pull request by spearw**
Modify structs. Refactor to allow for concurrency in downloading needed sections.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -324,13 +329,13 @@ pub struct SectionRegex {
...
-        bill_section: Regex::new(r"^(?i)section(?-i)\s*(\d*)\s*\.").unwrap(),
+        bill_section: Regex::new(r"^\s*(?i)section(?-i)\s*(\d*)\s*\.").unwrap(),
         amended: Regex::new(r"amended").unwrap(),
         striking: Regex::new(r"striking").unwrap(),
         inserting: Regex::new(r"inserting").unwrap(),
         repealed: Regex::new(r"repealed").unwrap(),
-        law_chapter: Regex::new(r"^(?i)section(?-i).*?[cC]hapter\s*(\d*\w*)").unwrap(),
-        law_section: Regex::new(r"^(?i)section(?-i).*?([sS]ection[s]?)\s*(\d*\w*)").unwrap(),
+        law_chapter: Regex::new(r"^\s*(?i)section(?-i).*?[cC]hapter\s*(\d*\w*)").unwrap(),
+        law_section: Regex::new(r"^\s*(?i)section(?-i).*?([sS]ection[s]?)\s*(\d*\w*)").unwrap(),
```
Comment (spearw) [inline src/lib.rs:338]: Probably should have left this for the regex PR, but this catches leading spaces before the section indicator

Comment (ralatsdc) [reply src/lib.rs:338]: Makes sense to do here.

File: src/lib.rs [hunk 2]
```diff
@@ -346,7 +351,7 @@ pub struct SectionCounts {
     pub other: i32,
 }
 
-pub fn init_section_counts() -> SectionCounts {
+fn init_section_counts() -> SectionCounts {
```
Comment (ralatsdc) [inline src/lib.rs:354]: Nice.

File: src/lib.rs [hunk 3]
```diff
@@ -529,29 +567,26 @@ pub fn get_law_section(law_chapter: &String, law_section: &String, tx: Sender<La
...
+        let body = reqwest::blocking::get(law_url.clone())
+            .unwrap()
+            .text()
+            .unwrap();
         let document = Html::parse_document(body.as_str());
 
         // Find the text node container
         let h2_selector = Selector::parse("h2#skipTo").unwrap();
-        let h2_element = document.select(&h2_selector).next().unwrap();
+        let h2_element = match document.select(&h2_selector).next() {
```
Comment (ralatsdc) [inline src/lib.rs:578]: Nice. We need to do this throughout (where sensible).

File: src/main.rs [hunk 4]
```diff
@@ -120,79 +120,48 @@ fn main() {
...
+            required_law_sections.dedup();
 
-            // Pop a vector of law location sections to use last
-            let mut law_location_sections = law_location.sections.clone();
-            let law_location_section = law_location_sections.pop().unwrap();
-            for law_location_section in law_location_sections {
-                get_law_section(&law_location.chapter, &law_location_section, tx.clone());
+            // Download required law sections concurrently
+            let (tx, rx) = mpsc::channel();
+            let chapter_section = required_law_sections.pop().unwrap();
```
Comment (ralatsdc) [inline src/main.rs]: Can we write this?
let (chapter, section) = required_law_sections.pop().unwrap();

Same if for.

**Review by ralatsdc — COMMENTED** [review]
I think the refactoring is excellent. I left a few comments though that you should read, and one that you should probably address.

**Pull request merged**